### PR TITLE
JS App examples PR

### PR DESCRIPTION
**Pull request by bparmar-splunk**
**Update:**

- App examples are migrated from JS SDK to app examples repo.
- Mod-inputs and browser examples are also migrated here.
- Automated tests using CI are added for JS examples except mod-input and browser.

### CODE REVIEW

File: javascript/node/README.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+> node search.js --search 'search index=_internal | head 20' | node results.js 
+
+[`results.js`] contains an example that shows how to handle result output from Splunk,
+including key-value extraction. It is meant to be used mostly in conjunction with [`search.js`]
+
+[`helloworld`]: https://github.com/splunk/splunk-sdk-javascript/tree/master/examples/node/helloworld
+[`jobs.js`]:    https://github.com/splunk/splunk-sdk-javascript/tree/master/examples/node/jobs.js
+[`conf.js`]:    https://github.com/splunk/splunk-sdk-javascript/tree/master/examples/node/conf.js
+[`search.js`]:  https://github.com/splunk/splunk-sdk-javascript/tree/master/examples/node/search.js
+[`results.js`]: https://github.com/splunk/splunk-sdk-javascript/tree/master/examples/node/results.js
```
Comment (fantavlik) [inline javascript/node/README.md:40]: These links should be updated to match this repo

File: javascript/node/helloworld/README.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+### Files: [`log.js`]
+
+This example shows how you can send data to Splunk over HTTP from within your 
+application by using the `Service.log` method in the JavaScript SDK.
+
+We create a utility `Logger` class that encapsulates various logging levels,
+and we can then simply call `logger.log`, `logger.error`, etc.
+
+[`apps.js`]:                    https://github.com/splunk/splunk-sdk-javascript/tree/master/examples/node/helloworld/apps.js
```
Comment (fantavlik) [inline javascript/node/helloworld/README.md:101]: These links need updating to match this repo

File: javascript/node/README.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
+## Node.js Examples
+
+This directory contains several examples of the Splunk Enterprise SDK for JavaScript running
```
Comment (fantavlik) [inline javascript/node/README.md:3]: Can we add guidance to install the required dependencies including the Splunk SDK for Javascript in the `./javascript` directory first by running `npm install`?

Comment (fantavlik) [reply javascript/node/README.md:3]: Actually I would suggest adding a README.md to the `./javascript` directory and populate it with all of the example related sections here: https://github.com/splunk/splunk-sdk-javascript/blob/5d600c1436d8b50954bc07b67db13e24312c92ef/README.md and also add the step above to instruct users to install with `npm install`. How does that sound to you?

**Review by fantavlik — CHANGES_REQUESTED** [review]
Added some review comments but this is looking great, glad to see examples running in CI as well!

File: javascript/node/README.md (hunk 3 above)
Comment (bparmar-splunk) [reply javascript/node/README.md:3]: Yes. It sounds correct to me for adding README inside javascript/ folder. 
But I am thinking of just adding a reference of app examples repo into Examples section of SDK repo README file. We will add reference links of app examples repo if user wants to test examples.

And we have planned changing all these link references as a separate task after migration.

**Review by fantavlik — APPROVED** [review]

**Pull request merged**